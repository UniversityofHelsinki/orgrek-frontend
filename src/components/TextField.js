import * as React from 'react';
import MuiTextField, {
  TextFieldProps as MuiTextFieldProps,
} from '@mui/material/TextField';
import Box from '@mui/material/Box';

const TextField = React.forwardRef((props, ref) => {
  const { onFocus, onBlur, helperText, ...other } = props;
  const maxLength = props.inputProps?.maxLength;

  const [visible, setVisible] = React.useState(false);

  return (
    <MuiTextField
      ref={ref}
      {...other}
      onFocus={(event) => {
        setVisible(true);
        onFocus && onFocus(event);
      }}
      onBlur={(event) => {
        setVisible(false);
        onBlur && onBlur(event);
      }}
      helperText={
        <Box
          component="span"
          sx={{ display: 'flex', justifyContent: 'space-between' }}
        >
          <span>{helperText}</span>
          {visible && maxLength > 0 && (
            <span>{`${props.value.length} / ${maxLength}`}</span>
          )}
        </Box>
      }
    />
  );
});

TextField.displayName = 'TextField';

export default TextField;
