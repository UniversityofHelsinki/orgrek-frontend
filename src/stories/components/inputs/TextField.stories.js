import React from 'react';
import TextField from '@mui/material/TextField';
import { useState } from 'react';

export default {
  component: TextField,
};

export const Basic = {
  args: {
    label: 'Label',
    value: '',
    helperText: '',
    required: false,
    disabled: false,
    error: false,
  },
  render: (args) => {
    const [value, setValue] = useState(args.value);

    return (
      <TextField
        {...args}
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const FormProps = {
  ...Basic,
  args: {
    ...Basic.args,
    value: 'Text field',
    helperText: 'Assistive text',
    required: true,
  },
};

export const Validation = {
  ...Basic,
  args: {
    ...Basic.args,
    value: 'Text field',
    helperText: 'Validation error message',
    required: true,
    error: true,
  },
};
