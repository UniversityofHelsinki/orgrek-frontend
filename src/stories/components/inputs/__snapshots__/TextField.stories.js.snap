// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/components/inputs/TextField Basic smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Label",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Label",
      "pressed": undefined,
      "role": "textbox",
      "value": undefined,
    },
  ],
  "name": "stories-components-inputs-textfield--basic",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/inputs/TextField FormProps smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Label",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "description": "Assistive text",
      "name": "Label",
      "pressed": undefined,
      "required": true,
      "role": "textbox",
      "value": "Text field",
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Assistive text",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "stories-components-inputs-textfield--form-props",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/inputs/TextField Validation smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Label",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "description": "Validation error message",
      "invalid": "true",
      "name": "Label",
      "pressed": undefined,
      "required": true,
      "role": "textbox",
      "value": "Text field",
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Validation error message",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "stories-components-inputs-textfield--validation",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
